history/history.c

```diff
@@ -0,0 +1,1081 @@
+/*
```

Comment:
Few generic comments for this file:
- Validate function args
- Use macros for sql queries
- Move useful macros to common place

It would be good to have a simple wrapper functions for `sqlite3_*`
function calls. If we want to migrate to a different db someday, we will have to change the code in one place which will make maintenance easy.

For example:
```
typedef sqlite3 db;

int db_open(const char *fn, db *_db)
{
    int rc;

    rc = sqlite3_open(fn, _db);
    return rc;
}

static db_init_db(const char *fn)
{
    int rc;
    db _db;

    rc = db_open(fn, &_db);
    check_db_rc(db, rc);

    return _db;
}
```
Like this, we need to have simple wrappers for each sqlite3 api.

Reply:
I don't think we should check args for every function call, that's just overkill. We can do that for the API calls, and I think I have done that, but not for every helper function.

As for alternative dbs, let's address that when needed, and I don't believe it will be soon. It will always be a local db, no point in using for example postgres. Besides, we have control over it. Even if RPM decides to use something else we do not need to switch.